src/delta_lake/delta_service.py

```diff
@@ -33,6 +35,57 @@
     "vacuum",
 }
 
+DISALLOW_SQL_META_CHARS = {
+    "--",
+    "/*",
+    "*/",
+    ";",
+    "\\",
+}
+
+ALLOWED_STATEMENTS = {
+    "select",
+}
+
+FORBIDDEN_POSTGRESQL_SCHEMAS = {
+    "pg_",
+    "pg_catalog",
+    "information_schema",
+}
+
+
+def _check_query_is_valid(query: str) -> bool:
+    """
+    Check if a query is valid.
+    """
+
+    statements = sqlparse.parse(query)
+    if len(statements) != 1:
+        raise SparkQueryError(f"Query {query} must contain exactly one statement")
+
+    statement = statements[0]
+    if statement.get_type().lower() not in ALLOWED_STATEMENTS:
+        raise SparkQueryError(
+            f"Query {query} must be one of the following: {', '.join(ALLOWED_STATEMENTS)}, got {statement.get_type()}"
+        )
+
+    if any(schema in query.lower() for schema in FORBIDDEN_POSTGRESQL_SCHEMAS):
+        raise SparkQueryError(
+            f"Query {query} contains forbidden PostgreSQL schema: {', '.join(FORBIDDEN_POSTGRESQL_SCHEMAS)}"
+        )
+
+    if any(char in query for char in DISALLOW_SQL_META_CHARS):
+        raise SparkQueryError(
+            f"Query {query} contains disallowed metacharacter: {', '.join(char for char in DISALLOW_SQL_META_CHARS if char in query)}"
+        )
+
+    if any(keyword in query.lower() for keyword in FORBIDDEN_KEYWORDS):
+        raise SparkQueryError(
+            f"Query {query} contains forbidden keyword: {', '.join(keyword for keyword in FORBIDDEN_KEYWORDS if keyword in query.lower())}"
+        )
```

Comment:
These can cause false positives; for example a column called `time_created` would trigger this error.

These are all statements, right? I would hope that if the code can check all the statements in the query and subqueries from the parsed sql then this would not longer be needed

Reply:
Good enough for now